apps/condo/domains/common/components/BaseSearchInput.tsx

```diff
@@ -0,0 +1,142 @@
+    const [fetching, setFetching] = useState(false)
+    const [data, setData] = useState([])
+    const [searchValue, setSearchValue] = useState('')
+    const [initialOptionsLoaded, setInitialOptionsLoaded] = useState(false)
+    const [initialValue, isInitialValueFetching] = useInitialValueGetter(restSelectProps.value, initialValueGetter)
+    const [scrollInputCaretToEnd, setSelectRef, selectInputNode] = useSelectCareeteControls(restSelectProps.id)
+
+    const searchSuggestions = useCallback(
+        async (value) => {
+            setFetching(true)
+            const data = await search((selected) ? selected + ' ' + value : value)
+
+            setFetching(false)
+            setData(data)
+        },
+        [],
+    )
+
+    const debouncedSearch = useMemo(
+        () => {
+            return debounce(searchSuggestions, DEBOUNCE_TIMEOUT)
+        },
+        [searchSuggestions],
+    )
+
+    const loadInitialOptions = useCallback(
+        (e) => {
+            if (props.onFocus) {
+                props.onFocus(e)
+            }
+
+            if (!initialOptionsLoaded) {
+                debouncedSearch(searchValue)
+                setInitialOptionsLoaded(true)
+            }
+        },
+        [initialOptionsLoaded],
+    )
+
+    const handleSelect = useCallback(
+        (value, option) => {
+            if (onSelect) {
+                props.onSelect(value, option)
+            }
+
+            setSelected(option.children)
+        },
+        [],
+    )
+
+    const handleClear = useCallback(
+        () => {
+            setSelected(null)
+        },
+        [],
+    )
+
+    useEffect(
+        () => {
+            setSearchValue(restSelectProps.value)
+        },
+        [restSelectProps.value]
+    )
+
+    useEffect(
+        () => {
+            setSearchValue(initialValue)
+        },
+        [initialValue],
+    )
+
+    useEffect(
+        () => {
+            scrollInputCaretToEnd(selectInputNode)
+        },
+        [selectInputNode, selected],
+    )
+
+    const options = useMemo(
+        () => data.map((option) => renderOption(option, restSelectProps.value)),
+        [data, fetching, restSelectProps.value],
+    )
+
+    return (
+        <Select
+            showSearch
+            autoFocus
+            allowClear
+            id={props.id}
+            value={searchValue}
+            disabled={Boolean(isInitialValueFetching)}
+            onFocus={loadInitialOptions}
+            onSearch={debouncedSearch}
+            onSelect={handleSelect}
+            onBlur={onBlur}
+            onChange={onChange}
+            onClear={handleClear}
+            ref={setSelectRef}
+            placeholder={placeholder}
+            notFoundContent={fetching ? <Spin size="small" /> : null}
+            // TODO(Dimitreee): remove ts ignore after combobox mode will be introduced after ant update
+            // @ts-ignore
+            mode={'SECRET_COMBOBOX_MODE_DO_NOT_USE'}
```

Comment:
please link to issue

Reply:
SECRET_COMBOBOX_MODE_DO_NOT_USE mode tmp deprecated.:(